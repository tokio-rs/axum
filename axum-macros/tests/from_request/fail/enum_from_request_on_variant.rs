use axum_macros::FromRequest;

#[derive(FromRequest, Clone)]
#[from_request(via(axum::Extension))]
enum Extractor {
    #[from_request(via(axum::Extension))]
    Foo,
}

fn main() {}
