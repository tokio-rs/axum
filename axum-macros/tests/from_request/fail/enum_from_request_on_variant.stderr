error: `#[from_request(via(...))]` cannot be used on variants
 --> tests/from_request/fail/enum_from_request_on_variant.rs:6:20
  |
6 |     #[from_request(via(axum::Extension))]
  |                    ^^^
